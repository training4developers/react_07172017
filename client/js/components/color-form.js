import React from 'react';

import { BaseForm } from './base-form';

export class ColorForm extends BaseForm {

  constructor(props) {
    super(props);

    this.state = {
      newColor: '',
    };
  }

  onClick = () => {
    this.props.onSaveColor(this.state.newColor);

    this.setState({
      newColor: '',
    });
  }

  render() {
    return <form>
      <div className="some-class">
        <label htmlFor="new-color-input">New Color:</label>
        <input type="text" id="new-color-input" name="newColor"
          value={this.state.newColor} onChange={this.onChange} />
      </div>
      <button type="button" onClick={this.onClick}>Add Color</button>
    </form>;
  }

}