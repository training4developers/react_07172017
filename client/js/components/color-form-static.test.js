import React from 'react';
import renderer from 'react-test-renderer';
import { render } from 'enzyme';

import { ColorForm } from './color-form';

describe('<ColorForm /> React Test Renderer Static HTML', () => {

  it('<ColorForm /> renders', () => {
    const tree = renderer.create(<ColorForm />).toJSON();
    expect(tree).toMatchSnapshot();
  });

});

describe('<ColorForm /> Enzyme Static HTML', () => {

  it('<ColorForm /> renders', () => {
    const component = JSON.stringify(render(<ColorForm />).html());
    expect(component).toMatchSnapshot();
  });

});
