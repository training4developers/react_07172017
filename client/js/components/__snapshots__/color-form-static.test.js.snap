// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ColorForm /> Enzyme Static HTML <ColorForm /> renders 1`] = `"\\"<form><div class=\\\\\\"some-class\\\\\\"><label for=\\\\\\"new-color-input\\\\\\">New Color:</label><input type=\\\\\\"text\\\\\\" id=\\\\\\"new-color-input\\\\\\" name=\\\\\\"newColor\\\\\\" value=\\\\\\"\\\\\\"></div><button type=\\\\\\"button\\\\\\">Add Color</button></form>\\""`;

exports[`<ColorForm /> React Test Renderer Static HTML <ColorForm /> renders 1`] = `
<form>
  <div
    className="some-class"
  >
    <label
      htmlFor="new-color-input"
    >
      New Color:
    </label>
    <input
      id="new-color-input"
      name="newColor"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <button
    onClick={[Function]}
    type="button"
  >
    Add Color
  </button>
</form>
`;
